File: Assignment 01/Program 02/prog2.py
# Write a C program to find the 3 rd /user defined position based maximum element in an array using dynamic memory allocation.

import array as arr

def bubble_sort(arr,n):
    for i in range(0,n):
        for j in range(0,n-i-1):
            if arr[j] < arr[j+1]:
                t = arr[j]
                arr[j] = arr[j+1]
                arr[j+1] = t


a = arr.array('i')

n = int(input("Enter the number of inputs:"))
for i in range(n):
    val = int(input(f"Enter the element {i+1}:"))
    a.append(val)

bubble_sort(a,n)

print("The 3rd position of the array based on the maximum element :",a[3])

File: Assignment 01/Program 02/prog2.c
// Write a C program to find the 3 rd /user defined position based maximum element in an array using dynamic memory allocation.

#include<stdio.h>
#include<stdlib.h>

void bubble_sort(int *arr, int n){
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < n-i-1; j++)
        {
            if(arr[j] < arr[j+1]){
                int temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
            }
        }
        
    }
    
}
int main(){
    int n;
    printf("Enter the number of inputs: ");
    scanf("%d",&n);
    int *arr;
    arr = (int*)malloc(n*sizeof(int));
    for (int i = 0; i < n; i++)
    {
        printf("Enter the element %d:",i+1);
        scanf("%d",(arr+i));
    }
    bubble_sort(arr,n);
    
    // for (int i = 0; i < n; i++)
    // {
    //     printf("%d ",*(arr+i));
    // }
    printf("The 3rd user defined position of the array based on max. element: %d",arr[3]);

    free(arr);
    return 0;
}